local/bin/py/build/content_manager.py

```diff
@@ -89,36 +91,55 @@ def grouped_globs_table(list_of_contents):
     return data
 
 
-def local_or_upstream(github_token, extract_dir, list_of_contents):
+def download_content_from_external_source(self, content):
+    """
+    Returns a boolean determining whether pull_config content should be downloaded from cache or external source.
+    Content will be pulled from external source if:
+        * cache is disabled
+        * generated file is not .md (this includes npm-integrations), which is currently unsupported.
+    """
+    file_name = content.get('options', {}).get('file_name', '')
+    action = content.get('action', '')
+
+    return (self.cache_enabled == False) \
+        or (action == 'npm-integrations') \
+        or (file_name != '' and not file_name.endswith('.md')) \
+        or (getenv("CI_COMMIT_REF_NAME") == 'master')
```

Comment:
Wanted to be extra cautious that we don't allow the cache building in `master` for now, just in case the flag is accidentally updated in the wrong `pull_config` file.